src/Pact/Eval.hs

```diff
@@ -748,13 +836,60 @@ resolveRef' _ _i (DName d@(DynamicName mem _ sigs i)) = do
   case a of
     Nothing -> evalError' i $ "resolveRef: dynamic ref not found: " <> pretty d
     Just r -> return $ Just r
+resolveRef' _ i _ = evalError' i $ "resolveRef: do not resolve fully qual names eagerly"
+
+-- | Variation of 'resolveRef' that allows nerfing the module ref attempt on bare names.
+resolveRefFQN :: HasInfo i => i -> Name -> Eval e (Maybe Ref)
+resolveRefFQN i (QName (QualifiedName q@(ModuleName refNs ns) n _)) = moduleResolver lookupQn i q
+  where
+    getModuleHash = \case
+      MDModule m -> pure (_mhHash $ _mHash m)
+      _ -> evalError' i $ "Qualified name access does not point to a module function " <> pretty q <> " " <> pretty n
+    lookupQn i' q' = do
+      m <- lookupModule i' q'
+      case (m, ns) of
+        (Just m', _) ->
+          case HM.lookup n (_mdRefMap m') of
+            Just (Ref TDef{}) -> do
+              h <- getModuleHash (_mdModule m')
+              let name' = FQName (FullyQualifiedName n q h)
+              return $ Just (Direct (TVar name' def))
+            p -> pure p
+        (Nothing, Just{}) -> return Nothing
+        (Nothing, Nothing) ->
+          -- note that while 'resolveModRef' uses 'moduleResolver' aTgain,
```

Comment:
```suggestion
          -- note that while 'resolveModRef' uses 'moduleResolver' again,
```

Reply:
iT wAS inTenTioNaL!!!

(mb tho will fix)